sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -518,9 +519,19 @@ lsc_block_load(const T *p, __ESIMD_NS::simd_mask<1> pred = 1) {
   constexpr detail::lsc_vector_size _VS =
       detail::to_lsc_vector_size<NElts / SmallIntFactor>();
   if constexpr (SmallIntFactor == 1) {
-    return __esimd_lsc_load_stateless<T, L1H, L3H, _AddressScale, _ImmOffset,
-                                      _DS, _VS, _Transposed, N>(pred.data(),
-                                                                addrs.data());
+    if constexpr (_DS == lsc_data_size::u32) {
+      __ESIMD_NS::simd<uint32_t, NElts> result =
+          __esimd_lsc_load_stateless<uint32_t, L1H, L3H, _AddressScale,
+                                     _ImmOffset, lsc_data_size::u32, _VS,
+                                     _Transposed, N>(pred.data(), addrs.data());
+      return result.template bit_cast_view<T>();
```

Comment:
Nit: I suggest to use `sycl::bit_cast` when view is not really needed (like in this case)